var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -374,14 +407,67 @@ def cmake_args(self):
                 '-DTPL_ENABLE_SuperLU:BOOL=OFF',
             ])
 
-        # python
-        if '+python' in spec:
+        if '+pnetcdf' in spec:
+            options.extend([
+                '-DTPL_ENABLE_Pnetcdf:BOOL=ON',
+                '-DTPL_Netcdf_Enables_Netcdf4:BOOL=ON',
+                '-DTPL_Netcdf_PARALLEL:BOOL=ON',
+                '-DPNetCDF_ROOT:PATH=%s' % spec['parallel-netcdf'].prefix
+            ])
+        else:
+            options.extend([
+                '-DTPL_ENABLE_Pnetcdf:BOOL=OFF'
+            ])
+
+        if '+zlib' in spec:
             options.extend([
-                '-DTrilinos_ENABLE_PyTrilinos:BOOL=ON'
+                '-DTPL_ENABLE_Zlib:BOOL=ON',
+                '-DZlib_ROOT:PATH=%s' % spec['zlib'].prefix,
             ])
         else:
             options.extend([
-                '-DTrilinos_ENABLE_PyTrilinos:BOOL=OFF'
+                '-DTPL_ENABLE_Zlib:BOOL=OFF'
+            ])
+
+        if '+belos' in spec:
+            options.extend([
+                '-DTrilinos_ENABLE_Belos:BOOL=ON'
+            ])
+
+        if '+zoltan' in spec:
+            options.extend([
+                '-DTrilinos_ENABLE_Zoltan:BOOL=ON',
+                '-DTrilinos_ENABLE_Zoltan2:BOOL=ON'
+            ])
+
+        if '+amesos' in spec:
+            options.extend([
+                '-DTrilinos_ENABLE_Amesos2:BOOL=ON',
+                '-DTrilinos_ENABLE_Amesos:BOOL=OFF'
+            ])
+
+        if '+muelu' in spec:
+            options.extend([
+                '-DTrilinos_ENABLE_MueLu:BOOL=ON'
+            ])
+
+        if '+ifpack' in spec:
+            options.extend([
+                '-DTrilinos_ENABLE_Ifpack2:BOOL=ON',
+                '-DTrilinos_ENABLE_Ifpack:BOOL=OFF'
+            ])
+
+        if '+tpetra' in spec:
+            options.extend([
+                '-DTrilinos_ENABLE_Tpetra:BOOL=ON',
+                '-DTpetra_INST_DOUBLE:BOOL=ON',
```

Comment:
I think instantiations should be a separate variant, like `eti` (ON by default). Then, you would write
```python
if '+eti' in spec:
    options.extend('-DTrilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON')
    if '+tpetra' in spec:
       # process Tpetra ETI
```